lib/server/services/dpl/getOrCreateDplDetector.js

```diff
@@ -11,12 +11,44 @@
  *  or submit itself to any jurisdiction.
  */
 
-const { DplDetectorRepository } = require('../../../database/repositories/index.js');
+const { sequelize } = require('../../../database/index.js');
+const { DplDetectorRepository, DetectorRepository } = require('../../../database/repositories/index.js');
+const { DetectorType } = require('../../../domain/enums/DetectorTypes.js');
 
 /**
  * Fetch (and create if it does not exist) a DPL detector
  *
  * @param {string} name the name of the dpl detector to find/create
  * @return {Promise<SequelizeDplDetector>} the dpl detector found
  */
-exports.getOrCreateDplDetectorByName = (name) => DplDetectorRepository.findOneOrCreate({ name });
+exports.getOrCreateDplDetectorByName = async (name) => {
+    let dplDetector = await DplDetectorRepository.findOne({ where: { name } });
+    if (dplDetector) {
+        return dplDetector;
+    } else {
+        /**
+         * Because of merging detectors and dpl_detector table,
+         * temporarly detectors and dpl_detectors with same name must also have same id
+         */
+        dplDetector = await DplDetectorRepository.insert({ name });
+        let detector = await DetectorRepository.findOne({ where: { name } });
+        if (!detector) {
+            detector = await DetectorRepository.insert({ name, type: DetectorType.QC });
+        }
+        const [[{ mid: maxDetectorId }]] = await sequelize.query('SELECT MAX(id) AS mid FROM detectors');
```

Comment:
I don't understand why you want a common id? Id is auto-generated, you should avoid setting it manually in any of the two tables.
You'll update the columns of tables pointing to DPL detectors when you'll create the fk for it, and you'll simply use the name as criteria, not id (because current id in dpl detectors do not match the ones in detectors)

Reply:
because: on frontend, only `detectorsProvider` remains, which will use `detectors` (not `dpl_detectors`) so e.g. QcFlag views will use identifiers of `detectors`, but still there are sql/sequelize views which use `dpl_detectors` and some FK constraints, in the migration you can see that all `detectors` have IDs copied from `dpl_detectors`, but in case something happened and next step cannot be taken soon, and still there is `dpl_detectors` table IDs in in both tables should still be the same. In order to avoid doing everything at once, this temporary solution allow this, also setting auto-generated id manually doesn't break anything

Reply:
I did not notice but in the migration you update the ids of detectors. This is not correct, because all the other tables that points to this will have wrong links now (for example: runs). You must not do that. In general, updating id is a very bad idea.

This probably mean that for now you can't delete the dplDetectors provider

Reply:
yes I can updated, because there is foriegn key constraint (onUpdate: CASCADE), and only reference to detectors table is in run_detectors table